### DISCUSSION TITLE
[Feature]: More compatibility mode?  Re: H/L remapping

### DISCUSSION THREAD

**Discussion by linsomniac** [LunarVim/LunarVim#1950] [Ideas]
### Feature motivation

I'm really liking the direction that LunarVim is going.  30 years of vi history under my belt, and LunarVim is a great config without having to manage my own configs and plugins.  But a recent change stomped on something I do all the time, and couldn't figure a way around:  I use H and L to move to the top and bottom lines of the screen ALL THE TIME.

I found in my config.lua I can add these lines:

    lvim.keys.normal_mode["<S-h>"] = nil
    lvim.keys.normal_mode["<S-l>"] = nil

to unmap the bindings, which works for me.

But I figured it's worth discussing the larger topic of how to handle diverging from "common vi mappings", which unfortunately would open the hair-splitting of "what exactly is a common mapping".  My common functions may not be anyone else's...  I could imagine having a "compatibility" mode that would avoid remapping common things.

Or possibly move "enhanced functionality" to Alt rather than Shift?  Kakoune seems to heavily use Alt.  I like the idea of h and l being used to move back and forth between tabs, but I have S-h and S-l as a frequent part of my workflow (probably dozens of times a day).

### Describe the expected solution

I'm not really sure, see discussion above.

### Describe the alternatives you've considered

Unmapping S-h and S-l in normal mode.  This is a fine solution for me, but begs a bigger question, especially when (as far as I could tell), there was no way to get the original S-h and S-l functionality.

C-e and C-y could sometimes be used instead, because one of my use patterns is S-l and then scroll down or S-h and then scroll up using the j/k keys.  But I also sometimes jump to the top or bottom of the window because that's closer to where I need to edit.

**Comment by chase** [comment]
Originally I think LunarVim was intended to be a pleasant transition from VS Code to neovim, so some of the default keybindings aren't very vim-like.

Recently I've made it possible to unset a default keybinding to the unbound behavior, so this shouldn't be as much of a problem: https://github.com/LunarVim/LunarVim/pull/1949#issuecomment-966862621

I have a keyboard with a very low pressure to activate so fairly frequently I would unintentionally trigger `<A-j/k>` while navigating through lines, so the default binding was annoying enough for me to resolve in a way that solves unsetting it in a generic reload-friendly manner.

That said, I do wonder if a more vim-like mode would be feasible with all of the plugins we include. Many of them set keymaps that LunarVim goes to great lengths to integrate, like luasnip, cmp, and autopairs.

**Reply by linsomniac** [reply]
I appreciate the change to allow removing a default binding.  I run into a similar problem to your <A-j/k>, in my case it's <ESC-j/k>, if I type them too quickly it moves the lines <A-j/k> style.  In my case, I have a custom keyboard layout that makes typing ESC quick, so I should track that down if it continues happening.

I understand that plugins may make a "more vi compatible" mode difficult, but wanted to open up the discussion.  I realize it may not be the direction LunarVim wants to take.  But Lunar really is a top tier vim setup and is my top contender for getting out of maintaining my own custom mappings and plugins that I've been pushing along for decades.

**Reply by polydevuk** [reply]
I have to say my otherwise glowing first impression of LunarVim was marred by the discovery that basic motions such as H and L had been removed as well as well as the default of h/l halting at the beginning/end of the line. Worse, even restoring these with the suggestions above has left L moving to the 8th row from the bottom. I cannot get L working as it does in Vim. I think prioritising the keymappings of plugins over core Vim motions is a very bad idea. Even VSCodeVim and IdeaVim don't go that far.

**Reply by rebuilt** [reply]
>  Worse, even restoring these with the suggestions above has left L moving to the 8th row from the bottom. I cannot get L working as it does in Vim.

You can make L do what you expect by changing `scrolloff`  (default: 0)
Add the following to your `~/.config/lvim/config.lua`
```
vim.opt.scrolloff = 0
```

**Reply by rebuilt** [reply]
Other options are also changed from their defaults by LunarVim.  They are listed here: 
https://github.com/LunarVim/LunarVim/blob/rolling/lua/lvim/config/settings.lua

**Reply by polydevuk** [reply]
Great. Just what I needed. Thanks.
